### ISSUE TITLE
(Vita) Frame Trottle is disable/not working

### ISSUE THREAD

**Issue by pralima87** [libretro/RetroArch#4461]
As of 1.40 Frame Trottle is still not working.
It was working in the early Vita releases back when Henkaku became popular but somehow it got disable.

**Comment by d30r** [comment]
I can verify that.
Vba-next core has this enabled. In all other cores I tested, Frame Throttling/Fast Forward is disabled.
Retro-Arch issue?

**Comment by pralima87** [comment]
Just tested 1.50 and slow motion works fine but FrameTrottle is still absent.

**Comment by francogp** [comment]
On my nvidia geforce GT 540m (with optimus), using GL, I can't Frame Throttling/Fast Forward. But on my Radeon R9 280x using vulkan and GL, I can.

**Comment by fr500** [comment]
Maybe you're forcing vsync on on NVIDIA control panel or it's forced on
because optimus..

**Comment by frangarcj** [comment]
It should be fixed with https://github.com/libretro/RetroArch/commit/1e7701bf29b446e3ab19ce47f35613121b0c206b

Could you check it @pralima87 ?

**Comment by pralima87** [comment]
yep S2!

**Comment by d30r** [comment]
@frangarcj Thank you so much , works like a charm!

**Comment by pralima87** [comment]
Fixed in 11.05.17 nighly!

**Comment by pralima87** [comment]
Issue is back in nighly 30/06/17

**Comment by frangarcj** [comment]
Do you know latest working version?

**Comment by pralima87** [comment]
Mr.frangarcj it was my mistake Frametrottle is working fine.
Is just the text "FrameTrottle" that is not showing up.

**Comment by meleu** [comment]
@twinaphex As you can see on the coment right above, this is not an issue anymore

**Comment by TomTsunami** [comment]
I've never been able to get Fast Forward to work either.  Have tried on a variety of cores.  If this hasn't been disabled I'm not sure what I'm doing wrong.  I've tried disabling Vsync.  I changed the Frame multiplier around (even though it says 0.0 will try to push to the max when you enable frame throttling).  I've tried buttons for Fast Forward Toggle Fast Forward Hold.  Nothing works.  Seemingly every other random option (take a screenshot, for example) will bind to a controller button, but not Fast Forward.  I hope someone can point me in the right direction.  I had installed RetroArch on my Vita a few months back.  I downloaded the news .vpk tonight.  No change.

**Comment by meleu** [comment]
@TomTsunami 
I simply configure the input to Fast Forward Hold (and Toggle) and it just works. No other config is needed.

If you think it can help, here is my Vita's [retroarch.cfg](https://github.com/libretro/RetroArch/files/1858434/retroarch.cfg.txt).

**Comment by TomTsunami** [comment]
Thanks for sending that.  I checked it against mine - don't see any discrepancies.  Here is mine.  I've tried all the SNES cores, PCE, and FCEU.  Can't get it to work in any of them.
[retroarch_me.cfg.txt](https://github.com/libretro/RetroArch/files/1858474/retroarch_me.cfg.txt)

**Comment by TomTsunami** [comment]
Is it possible I have an outdated version of RetroArch?  My virgin install was a few months back.  I downloaded the latest .vpk yesterday and installed from VitaShell but I don't know how well that garbage collects/reinstalls.  Not sure where to find version number inside RetroArch, either.  It seems like at one time this was just disabled as a feature?

**Comment by meleu** [comment]
suggestion:
- uninstall/delete RetroArch (keep the RetroArch's bubble pressed and then select "Delete").
- remove the directory `ux0/data/retroarch/`.

and then reinstall everything again using nightly.

**Comment by TomTsunami** [comment]
thank you meleu, your suggestion worked.

I will say, unpacking/installing retroarch.vpk without uninstalling through the bubble must not work right at all.  The version of RetroArch I am running now is significantly different looking than whatever I was running.  Also, it took about 30 seconds to unpack/install through vitahsell whereas the other day it took about 45 minutes.  The fast forward now works.  I get the notification.  Interestingly, at first I didn't think it was actually doing anything.  So I turned on Show FPS and learned that some of the cores can't really be sped up much by the Vita.
Using Chrono Trigger as a test case, here are the ballpark frame rates I got with fast forward toggled on:

SNES9x 2010: Usually it has no impact as the game, whether FF is on or off can't crack 60 fps.
SNES9x 2005+: Up to low 70s
SNES9x 2005: 80-110
SNES9x 2002: 100-130

I tried Tales of Phantasia as well, with similar results:
SNES9x 2010: Usually it has no impact as the game, whether FF is on or off can't crack 60 fps.
SNES9x 2005+: 55-100
SNES9x 2005: 60-130
SNES9x 2002: 70-150

So if you are looking to grind through those JRPGS definitely use the 2002 core.
